src/bin/inx-chronicle/stardust_inx/mod.rs

```diff
@@ -275,6 +298,19 @@ impl InxWorker {
                 .collection::<ProtocolUpdateCollection>()
                 .insert_protocol_parameters(start_index, protocol_parameters.into())
                 .await?;
+
+            // Get the latest milestone again after reading unspent outputs
+            let starting_index = inx
```

Comment:
Is this index required to be the index for which we requested the unspent outputs? Or asked in a different way: if we got the unspent outputs at index A ... does it matter that this call could return some index A+x (x!=0)?

Reply:
It's supposed to be, but since we have to make two calls, it's possible for it to be the next milestone, which is better than it being the previous one.

Reply:
Why don't we use the index returned by `read_unspent_outputs`? Isn't that guaranteed to be the correct one?

Reply:
Depends. Is the `ledger_index` on `UnspentOutputMessage` the latest index, or is it the index of the output?

Reply:
It's the latest index, it tells us when the output was still unspent. The only thing we might need to check if all returned ledger indexes are the same (which they should be, because Hornet locks the ledger afaik).

Reply:
It won't be very clean to extract that value from the stream but it's doable. I guess the question is: should we?

Reply:
I think so, and we should only set the starting index if `read_unspent_output` was sucessful and all returned outputs have the same ledger index.